src/audio/codec_adapter/codec_adapter.c

```diff
@@ -306,7 +306,16 @@ int codec_adapter_params(struct comp_dev *dev,
 		return ret;
 	}
 
-	ret = memcpy_s(&mod->stream_params, sizeof(struct sof_ipc_stream_params),
+	/* allocate stream_params each time */
```

Comment:
I feel like this is kinda wasteful and error-prone. I'd say allocate first time, reuse on further uses.

Reply:
We could go with that too, but the problem would be if kernel decides to change ext_data_len between calls. Anyway, codec_adapter_params is called only once.

Reply:
You're always allocating with a compile time size, sizeof(struct ...). Which is why I suggested this in the first place.

Reply:
@paulstelian97 this changes in the 4th patch. I preferred to split the code this way to be easier to understand.

```
mod->stream_params = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM,
				     sizeof(*mod->stream_params));
				     sizeof(*mod->stream_params) + params->ext_data_length);
```